dll/cpl/sysdm/environment.c

```diff
@@ -45,97 +193,788 @@ GetSelectedListViewItem(HWND hwndListView)
+
+    pidllist = SHBrowseForFolder(&bi);
+    if (!pidllist)
+    {
+        return;
+    }
+
+    if (SHGetPathFromIDList(pidllist, szDir))
+    {
+        if (DlgData->bIsFriendlyUI)
+        {
+            /* If no item is selected then create a new empty item and add the required location to it */
+            if (!DlgData->bIsItemSelected)
+            {
+                DlgData->dwSelectedValueIndex = ListView_GetItemCount(hwndListView);
+                AddEmptyItem(hwndListView, DlgData->dwSelectedValueIndex);
+            }
+            ListView_SetItemText(hwndListView,
+                                 DlgData->dwSelectedValueIndex,
+                                 0,
+                                 szDir);
+            ListView_SetItemState(hwndListView, DlgData->dwSelectedValueIndex,
+                                  LVIS_FOCUSED | LVIS_SELECTED,
+                                  LVIS_FOCUSED | LVIS_SELECTED);
+        }
+        else
+        {
+            SetDlgItemText(hwndDlg, IDC_VARIABLE_VALUE, szDir);
+        }
+    }
+
+    CoTaskMemFree(pidllist);
+}
+
+static VOID
+MoveListItem(HWND hwndDlg,
+             PEDIT_DIALOG_DATA DlgData,
+             BOOL bMoveUp)
+{
+    TCHAR szDest[MAX_PATH];
+    TCHAR szSource[MAX_PATH];
+    HWND hwndListView;
+    DWORD dwSrcIndex, dwDestIndex, dwLastIndex;
+
+    hwndListView = GetDlgItem(hwndDlg, IDC_LIST_VARIABLE_VALUE);
+
+    dwLastIndex = ListView_GetItemCount(hwndListView) - 1;
+    dwSrcIndex = DlgData->dwSelectedValueIndex;
+    dwDestIndex = bMoveUp ? (dwSrcIndex - 1) : (dwSrcIndex + 1);
+
+    if ((bMoveUp && dwSrcIndex > 0) || (!bMoveUp && dwSrcIndex < dwLastIndex))
+    {
+        ListView_GetItemText(hwndListView,
+                             dwSrcIndex,
+                             0,
+                             szDest,
+                             _countof(szDest));
+        ListView_GetItemText(hwndListView,
+                             dwDestIndex,
+                             0,
+                             szSource,
+                             _countof(szSource));
+
+        ListView_SetItemText(hwndListView,
+                             dwDestIndex,
+                             0,
+                             szDest);
+        ListView_SetItemText(hwndListView,
+                             dwSrcIndex,
+                             0,
+                             szSource);
+
+        DlgData->dwSelectedValueIndex = dwDestIndex;
+        ListView_SetItemState(hwndListView, DlgData->dwSelectedValueIndex,
+                              LVIS_FOCUSED | LVIS_SELECTED,
+                              LVIS_FOCUSED | LVIS_SELECTED);
+    }
+}
+
+static VOID
+OnEnvironmentEditDlgResize(HWND hwndDlg,
+                           PEDIT_DIALOG_DATA DlgData,
+                           DWORD cx,
+                           DWORD cy)
+{
+    RECT rect;
+    HDWP hdwp = NULL;
+    HWND hItemWnd;
+
+    if ((cx != DlgData->cxOld) || (cy != DlgData->cyOld))
```

Comment:
I would suggest, in order to remove one level of indentation, to do this:
```suggestion
    if ((cx == DlgData->cxOld) && (cy == DlgData->cyOld))
        return;
```
then comes the actual resizing code.